### DISCUSSION TITLE
Does the game in any way prevent malicious rulesets?

### DISCUSSION THREAD

**Discussion by SolsticeSpectrum** [ppy/osu#28060] [Ideas] [answered]
There is one closed source ruleset called Yoso and it got me wondering. Are there systems in place to prevent the ruleset from acting evil? Some sort of built-in security measure should be a thing.

**Comment by LumpBloom7** [comment]
At the moment there are no direct measures in place, and I don't think that it is possible to effectively prevent such a scenario since rulesets are not limited to the API that osu provides.  

It goes without saying that you shouldn't download something if you don't trust the origin.

I don't believe Yoso is malicious (or any other currently available rulesets for that matter).

**Reply by SolsticeSpectrum** [reply]
Well at least it shouldn't be DLL files. DLL files are really dangerous in nature.

**Reply by peppy** [reply, marked as answer]
Rulesets are basically games of their own. As already mentioned above, you should trust the source before installing any non-official rulesets.

**Reply by bdach** [reply]
> DLL files are really dangerous in nature

So are EXE files for that matter by the way. Which ties into the above again really.

**Reply by SolsticeSpectrum** [reply]
that's why something else might be worth it, like a custom file format for this, it would prevent arbitrary code to be executed. I mean if an RCE was found, then it wouldn't but it would make it harder to just put some funky code in the DLL. Maybe a `.ruleset` file format?

**Reply by bdach** [reply]
That is monumental effort that doesn't even give any real guarantees because any theoretical format we put in place will eventually be executable in some form and thus potentially exploitable. If you can't trust it then run it in a sandbox. Or not at all.

**Reply by peppy** [reply]
Yep, out of scope of this project.
